### PR TITLE
Update

### PR DESCRIPTION
**Pull request by anmyachev**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,13 +6,19 @@ Based on [benchmarks published by NVidia for their RAPIDS](https://render.github
...
 - follow [the steps](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/install-nvidia-driver.html#Cluster_GPUs_Manual_Install_Driver) to install the necessary driver for V100 GPU (using 396.82 version)
-- `git clone https://github.com/vnlitvin/mortgage-benchmarks.git`
-- `cd mortgage-benchmarks`
-- `chmod +x test_mortgage_gpu.sh && ./test_mortgage_gpu.sh`
-- `bash`
-- `conda activate cudf_dev`
-- `pip install xgboost`
-- `time python Mortgage_GPU.py  test_mortgage_gpu/dataset/ 1`
+- `git clone https://github.com/vnlitvin/mortgage-benchmarks.git && cd mortgage-benchmarks`
+- `git checkout amyachev`
```
Comment (vnlitvinov) [inline README.md]: is this needed after the merge? seems not

File: README.md [hunk 2]
```diff
@@ -6,13 +6,19 @@ Based on [benchmarks published by NVidia for their RAPIDS](https://render.github
...
+- `chmod +x run_gpu.sh && . ./run_gpu.sh`
```
Comment (vnlitvinov) [inline README.md]: all shell scripts should be made executable in `git` so as to not require being `chmod`-ed in place

File: README.md [hunk 3]
```diff
@@ -6,13 +6,19 @@ Based on [benchmarks published by NVidia for their RAPIDS](https://render.github
...
+- `time python Mortgage_GPU.py  test_mortgage_gpu/mortgage_dataset/ 1`
+
+## CPU case with using pandas upstream
+- launch `c5.18xlarge` cpu instance on AWS cloud with `Ubuntu Server 18.04 LTS (HVM), SSD Volume Type` image
+- login in it
+- `git clone https://github.com/vnlitvin/mortgage-benchmarks.git && cd mortgage-benchmarks`
+- `git checkout amyachev`
+- `chmod +x run_cpu.sh && . ./run_cpu.sh`
```
Comment (vnlitvinov) [inline README.md]: same on two lines

File: prereq_install.sh [hunk 4]
```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+
+# functions
+function create_dir {
```
Comment (vnlitvinov) [inline prereq_install.sh]: wtf? why don't you just use `mkdir -p $target_dir`?..

File: prereq_install.sh [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+sudo apt-get update -y
+sudo apt-get install gcc g++ -y
+
+
+create_dir $TEST_FOLDER
+cd $TEST_FOLDER
+
+
+if [[ ! "$(command -v conda)" ]]; then
+    echo "conda command wasn't found in the PATH; attemp to install miniconda"
```
Comment (vnlitvinov) [inline prereq_install.sh]:
```suggestion
    echo 'conda command wasn't found in $PATH; attempting to install miniconda to "~/miniconda"'
```

File: prereq_install.sh [hunk 6]
```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
```
Comment (vnlitvinov) [inline prereq_install.sh]: you should `set -e` IIRC to exit the script on any non-zero exit code of a command ("stop on error" thing)

File: prereq_install.sh [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+if [[ ! "$(command -v conda)" ]]; then
+    echo "conda command wasn't found in the PATH; attemp to install miniconda"
+    wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
+    bash Miniconda3-latest-Linux-x86_64.sh -b -p $HOME/miniconda
+    rm Miniconda3-latest-Linux-x86_64.sh
+    export PATH="$HOME/miniconda/bin:$PATH"
+    conda init bash
+    source ~/.bashrc
+    echo "conda installed"
```
Comment (vnlitvinov) [inline prereq_install.sh]:
```suggestion
    echo "conda installed to ~/miniconda"
```

File: prereq_install.sh [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+else
+    echo "conda command was found in the PATH"
```
Comment (vnlitvinov) [inline prereq_install.sh]:
```suggestion
    echo 'conda found in $PATH'
```

File: prereq_install.sh [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+echo ""
+
+
+create_dir $DATASET_FOLDER
+cd $DATASET_FOLDER
+
+
+if [[ ! -f $DATASET_NAME ]]; then
+    echo "$DATASET_NAME file wasn't found"
+    echo "attemp to download this file from $DATASET_LINK"
```
Comment (vnlitvinov) [inline prereq_install.sh]:
```suggestion
    echo "downloading it from $DATASET_LINK"
```

File: prereq_install.sh [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+    wget $DATASET_LINK
+    tar -xzvf $DATASET_NAME
+else
+    echo "$DATASET_NAME file was found"
+fi
+cd ../..
```
Comment (vnlitvinov) [inline prereq_install.sh:49]: this is usually useless, as working directory is reset after script is invoked. I see that you're source-ing it later, so it' not useles here, but if you want to preserve directories use `pushd/popd` stuff

File: requirements_cpu_daal.yml [hunk 11]
```diff
@@ -0,0 +1,13 @@
+name: mortgage_cpu_daal
+channels:
+- conda-forge
+- intel
+- defaults
+dependencies:
```
Comment (vnlitvinov) [inline requirements_cpu_daal.yml]: why there's no dependency on `pandas>=0.25`?

File: run_cpu.sh [hunk 12]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+
+# prepare conda tool, mortgage dataset
+. ./prereq_install.sh
+
+conda env create -f requirements_cpu_daal.yml
+echo "env mortgage_cpu_daal is created"
+
+echo "conda activate mortgage_cpu_daal"
```
Comment (vnlitvinov) [inline run_cpu.sh]: eh, what is this? why is this even needed?

File: run_cpu.sh [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+conda activate mortgage_cpu_daal
+echo "mortgage_cpu_daal is activated"
+
+# build pandas from source
```
Comment (vnlitvinov) [inline run_cpu.sh]: this should be no longer needed as they've finally released 0.25

File: README.md (hunk 3 above)
Comment (vnlitvinov) [inline README.md]: btw both `run_cpu` and `run_gpu` script names look misleading as they aren't actually _running_ the benchmark, it's more like `prepare_cpu` and `prepare_gpu` IMHO

File: run_cpu.sh [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+git clone https://github.com/pandas-dev/pandas.git && cd pandas
+
+echo "building pandas from master"
+python setup.py install
+echo "build successful"
+
+cd ..
```
Comment (vnlitvinov) [inline run_cpu.sh]: I would suggest `echo`-ing the very short instruction on how to run the benchmark (like "now you can run 'time python blah/script.py foo 1'")

File: run_gpu.sh [hunk 15]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+. ./prereq_install.sh
+
+conda env create -f ./requirements_gpu.yml
+conda activate mortgage_gpu
```
Comment (vnlitvinov) [inline run_gpu.sh:6]: same on short instructions

File: README.md (hunk 1 above)
Comment (anmyachev) [reply README.md]: fixed

File: README.md (hunk 2 above)
Comment (anmyachev) [reply README.md]: done

File: prereq_install.sh (hunk 4 above)
Comment (anmyachev) [reply prereq_install.sh]: replaced

File: run_cpu.sh (hunk 13 above)
Comment (anmyachev) [reply run_cpu.sh]: We no longer want to get the actual performance numbers?

File: README.md (hunk 3 above)
Comment (anmyachev) [reply README.md]: I agree. Fixed

File: README.md [hunk 16]
```diff
@@ -10,15 +10,13 @@ Based on [benchmarks published by NVidia for their RAPIDS](https://render.github
...
+- `. ./run_gpu.sh`
 - `time python Mortgage_GPU.py  test_mortgage/mortgage_dataset/ 1`
 
 ## CPU case with using pandas upstream
 - launch `c5.18xlarge` cpu instance on AWS cloud with `Ubuntu Server 18.04 LTS (HVM), SSD Volume Type` image
 - login in it
 - `git clone https://github.com/vnlitvin/mortgage-benchmarks.git && cd mortgage-benchmarks`
-- `git checkout amyachev`
-- `chmod +x run_cpu.sh && . ./run_cpu.sh`
+- `. ./run_cpu.sh`
```
Comment (vnlitvinov) [inline README.md]: this is outdated, too

File: README.md [hunk 17]
```diff
@@ -10,15 +10,13 @@ Based on [benchmarks published by NVidia for their RAPIDS](https://render.github
 - login in it
 - follow [the steps](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/install-nvidia-driver.html#Cluster_GPUs_Manual_Install_Driver) to install the necessary driver for V100 GPU (using 396.82 version)
 - `git clone https://github.com/vnlitvin/mortgage-benchmarks.git && cd mortgage-benchmarks`
-- `git checkout amyachev`
-- `chmod +x run_gpu.sh && . ./run_gpu.sh`
+- `. ./run_gpu.sh`
```
Comment (vnlitvinov) [inline README.md]: this line is outdated now

File: prereq_install.sh (hunk 6 above)
Comment (vnlitvinov) [reply prereq_install.sh]: what about `set -e` here and in other scripts?

File: README.md (hunk 16 above)
Comment (anmyachev) [reply README.md]: fixed

File: README.md (hunk 17 above)
Comment (anmyachev) [reply README.md]: fixed

**Comment by anmyachev** [comment]
@vnlitvin need to test with this option

**Comment by vnlitvinov** [comment]
> @vnlitvin need to test with this option

pls do :)

File: prereq_install.sh (hunk 6 above)
Comment (anmyachev) [reply prereq_install.sh]: done

File: run_gpu.sh [hunk 18]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+set -e
+. ./prereq_install.sh
+
+conda env create -f ./requirements_gpu.yml
```
Comment (vnlitvinov) [inline run_gpu.sh]: pls follow pattern from `run_cpu.sh` - use `source` instead of `.` and do an existing environment check

**Review by vnlitvinov — CHANGES_REQUESTED** [review]

File: run_gpu.sh (hunk 18 above)
Comment (anmyachev) [reply run_gpu.sh]: fixed

File: requirements_cpu_daal.yml [hunk 19]
```diff
@@ -0,0 +1,12 @@
+name: mortgage_cpu_daal
```
Comment (vnlitvinov) [inline requirements_cpu_daal.yml]: looks like the file is named wrong, as it's both for "cpu_daal" and "cpu_xgboost" cases

Comment (anmyachev) [reply requirements_cpu_daal.yml]: fixed

**Review by vnlitvinov — APPROVED** [review]

**Pull request merged**